### PR TITLE
Fix bug in filegeneration editor not resetting while changing elements

### LINKED ISSUE #142: Bug: File Generation Editor does not reset when changing elements
### Step to Reproduce

Create two classes (Class A, Class B). 
Open the generated view for Class A to avro. 
Open the generated view for Class B to avro. 
File generation editor for class b will not auto open the avro generated file (it will for class A). 

### Expected behavior

When opening the generated view for Class B, the file generation editor should open the first avro generated file.

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary
Closes #142 

## How did you test this change?

https://user-images.githubusercontent.com/87973126/141459981-c43317f7-4221-456c-84f1-503e12df3b4a.mp4

### CODE REVIEW

File: .changeset/clever-seas-study.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@finos/legend-studio": patch
+---
+
+Fix bug in filegeneration editor not resetting while changing elements
```
Comment (MauricioUyaguari) [inline .changeset/clever-seas-study.md:5]:
````suggestion
Fix bug in `FileGenerationEditor` not resetting while changing elements (see [#142](https://github.com/finos/legend-studio/issues/142)] for more details).
````

File: packages/legend-studio/src/stores/editor-state/FileGenerationState.ts [hunk 2]
```diff
@@ -180,11 +180,10 @@ export class FileGenerationState {
     reprocessOpenNodes(treeData, this.filesIndex, this.root, openedNodeIds);
     // select the current file node if available, else select the first output
     const selectedFileNodePath =
-      this.selectedNode?.fileNode.path ??
-      (generationResult.length === 1
+      generationResult.length === 1
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/editor-state/FileGenerationState.ts:183]: This is moving in the right direction but we are not reopening the same file in a regenerate that the user had opened instead we are always opening the first element in the generation output. 
I think we should follow the below logic. 
if `this.selectedNode?.fileNode.path` exists and the corresponding file exists in the file index open that. 
If not open the first generationOutput if there is one.

**Comment by MauricioUyaguari** [comment]
I think we are close with this change, however we don't handle processing of file generations with no generation results. 
Use the grammar below. If you open the file generation editor, you will see an error. So when there is no selectedNode we should only set the selectedNode to the first generation result if there are results. 
```
###Text
Text model::A
{
  type: plainText;
  content: '';
}


###FileGeneration
Avro model::MyFilegeneration
{
  scopeElements: [model];
}

```

**Comment by MauricioUyaguari** [comment]
Appreciate the bug fix @gayathrir11 . Could we check one more thing? Let's investigate how we can add for tests for this. Im okay with filing a separate issue for tests if you think it will be too involved and leave it for another commit.

**Review by MauricioUyaguari — COMMENTED** [review]
Left some comments!

**Comment by gayathrir11** [comment]
> Appreciate the bug fix @gayathrir11 . Could we check one more thing? Let's investigate how we can add for tests for this. Im okay with filing a separate issue for tests if you think it will be too involved and leave it for another commit.

I think we should leave it for another commit

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**